rating/peersRatingHandler.go

```diff
@@ -232,6 +256,39 @@ func (prh *peersRatingHandler) splitPeersByTiers(peers []core.PeerID) ([]core.Pe
 	return topRated, badRated
 }
 
+func (prh *peersRatingHandler) updateRatingsMap(pid core.PeerID, newRating int32, updateFactor int32) {
+	prettyPID := pid.Pretty()
+	peerRatingInfo, exists := prh.ratingsMap[prettyPID]
+	if !exists {
+		prh.ratingsMap[prettyPID] = &ratingInfo{
+			Rating:                       newRating,
+			TimestampLastRequestToPid:    0,
+			TimestampLastResponseFromPid: 0,
+		}
+		return
+	}
+
+	peerRatingInfo.Rating = newRating
+
+	newTimeStamp := prh.getTimeHandler().Unix()
+	if updateFactor == decreaseFactor {
+		peerRatingInfo.TimestampLastRequestToPid = newTimeStamp
+		return
+	}
+
+	peerRatingInfo.TimestampLastResponseFromPid = newTimeStamp
+}
+
+func (prh *peersRatingHandler) updateMetrics() {
```

Comment:
This will make the whole component a lot slower
Why not call this on a separate go routine set at a predefined time period? Say 1 minute?

Reply:
added go routine